src/openrct2/core/Guard.cpp

```diff
@@ -78,15 +78,11 @@ namespace Guard
         Console::Error::WriteLine("Version: %s", gVersionInfoFull);
 
         // This is never freed, but acceptable considering we are about to crash out
-        utf8* formattedMessage = nullptr;
+        std::string formattedMessage;
         if (message != nullptr)
         {
             formattedMessage = String::Format_VA(message, args);
-            Console::Error::WriteLine(formattedMessage);
-        }
-
-        if (formattedMessage != nullptr)
-        {
+            Console::Error::WriteLine(formattedMessage.data());
```

Comment:
As stated multiple times now already, use `c_str` not `data`. `c_str` is guaranteed to be null-terminated while `data` is not. And also only do that if you really have to. Preferably, you do not call either of those functions.

Reply:
Addressed by switching data to c_str.